tests/dialects/test_duckdb.py

```diff
@@ -753,13 +753,15 @@ def test_duckdb(self):
             },
         )
 
-        with self.assertRaises(UnsupportedError):
-            transpile(
-                "SELECT REGEXP_EXTRACT(a, 'pattern', 1) from table",
-                read="bigquery",
-                write="duckdb",
-                unsupported_level=ErrorLevel.IMMEDIATE,
-            )
```

Comment:
Am I misunderstanding this, doesn't this PR fix this so this case, which previously failed, should now pass? Or if this should still be an xfail, can you explain why?

Reply:
From what I understood your PR is adding support for DuckDB's options/parameters such as case sensitive/insensitive matching. This doesn't seem to be supported by BigQuery's signature:

```
REGEXP_EXTRACT(value, regexp[, position[, occurrence]])
```

And vice versa, DuckDB doesn't support `position` and `occurence`. 

For this reason the generator exposes a couple tools (`self.unsupported(msg)` and `@unsupported_args` decorator) which should be used to warn the user that if they use these arguments (e.g `position` in BigQuery) it's not possible to transpile them to DuckDB; This is what this case is testing too, for a higher unsupported level (error instead of warning)

Reply:
In contrast, this is Snowflake's signature:

```
REGEXP_SUBSTR( <subject> , <pattern> [ , <position> [ , <occurrence> [ , <regex_parameters> [ , <group_num> ] ] ] ] )
```

The `regex_parameters` here are similar to DuckDB's `parameters`, so transpiling that specific argument between DuckDB <-> Snowflake is fine after your PR

Reply:
ah, sorry, I though BQ's signature was `REGEXP_EXTRACT(value, regexp[, occurrence])`, but now I say that, I don't see how this PR would affect whether or not that works. Thanks.